framework/src/auxkernels/VariableOldValueBoundsAux.C

```diff
@@ -0,0 +1,36 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VariableOldValueBoundsAux.h"
+
+registerMooseObject("MooseApp", VariableOldValueBoundsAux);
+
+defineLegacyParams(VariableOldValueBoundsAux);
+
+InputParameters
+VariableOldValueBoundsAux::validParams()
+{
+  InputParameters params = BoundsAuxBase::validParams();
+  params.addClassDescription("Provides the upper and lower bound of the phase field fracture "
+                             "variable to PETSc's SNES variational inequalities solver.");
+  return params;
+}
+
+VariableOldValueBoundsAux::VariableOldValueBoundsAux(const InputParameters & parameters)
+  : BoundsAuxBase(parameters)
+{
+}
+
+Real
+VariableOldValueBoundsAux::computeBound()
+{
+  MooseVariable & var = _subproblem.getStandardVariable(_tid, _bounded_var_name);
```

Comment:
I don't know how much cost this is, but shouldn't this go into the initialization list? I wouldn't expect the variable to change during simulation.

Reply:
Changed.